plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtFenceCommandWrapper.java

```diff
@@ -47,20 +48,25 @@ public Answer execute(final FenceCommand command, final LibvirtComputingResource
         final ExecutorService executors = Executors.newSingleThreadExecutor();
         final KVMHAMonitor monitor = libvirtComputingResource.getMonitor();
 
-        final List<NfsStoragePool> pools = monitor.getStoragePools();
+        final List<NfsStoragePool> nfspools = monitor.getStoragePools();
+        final List<RbdStoragePool> rbdpools = monitor.getRbdStoragePools();
 
         /**
          * We can only safely fence off hosts when we use NFS
          * On NFS primary storage pools hosts continuesly write
          * a heartbeat. Disable Fencing Off for hosts without NFS
          */
-        if (pools.size() == 0) {
+        if (nfspools.size() == 0) {
             String logline = "No NFS storage pools found. No way to safely fence " + command.getVmName() + " on host " + command.getHostGuid();
             s_logger.warn(logline);
             return new FenceAnswer(command, false, logline);
+        }else if (rbdpools.size() == 0) {
+            String logline = "No RBD storage pools found. No way to safely fence " + command.getVmName() + " on host " + command.getHostGuid();
```

Comment:
We could use `String.format` here.

Reply:
@GutoVeronezi    I modified the code as you requested.